File: kotlin/src/main/kotlin/TextNote.kt
import java.time.Instant

class TextNote(
    data: TextNoteModel,
    storage: Storage<TextNoteModel>,
    dateFormatter: DateFormatting
) : Note<TextNoteModel>(data, storage, dateFormatter), AnyNote {

    override fun drawCanvas(): String {
        val m = storage.valueById(id) ?: data
        return """
        [Текстовая заметка]
        • id: $id
        • название: ${m.name}
        • текст заметки: ${m.text}
        • создано: ${dateFormatter.format(m.createdAt)}
        • изменено: ${dateFormatter.format(m.updatedAt)}
        """.trimIndent()
    }

    fun rename(newName: String) {
        val m = (storage.valueById(id) ?: data).copy(name = newName, updatedAt = Instant.now())
        update(m)
    }

    fun updateText(newText: String) {
        val m = (storage.valueById(id) ?: data).copy(text = newText, updatedAt = Instant.now())
        update(m)
    }
}

File: swift/Sources/Notes/TextNote.swift
import Foundation

final class TextNote: Note<TextNoteModel>, AnyNote {
    override func drawCanvas() -> String {
        let m = storage.value(by: id) ?? data
        return """
        [Текстовая заметка]
        • id: \(id)
        • название: \(m.name)
        • текст заметки: \(m.text)
        • создано: \(dateFormatter.format(m.createdAt))
        • изменено: \(dateFormatter.format(m.updatedAt))
        """
    }

    func rename(_ newName: String) {
        var m = storage.value(by: id) ?? data
        m.name = newName
        m.updatedAt = Date()
        update(m)
    }

    func updateText(_ newText: String) {
        var m = storage.value(by: id) ?? data
        m.text = newText
        m.updatedAt = Date()
        update(m)
    }
}
